src/OrgasmControl.cpp

```diff
@@ -95,6 +119,47 @@ namespace OrgasmControl {
         Hardware::setMotorSpeed(motor_speed);
       }
     }
+    
+    void updateEdgingTime() {
+      if (!control_motor) {                   // keep edging start time to current time as long as system is in Manual Mode
+        autoEdgingStartMillis = millis();
+        postOrgasmStartMillis = 0;
+        if (Config.sensitivity_threshold == 6000) { // set back the sensitivity_threshold if switched to manual mode before the end of the post orgasm cycle
+          Config.sensitivity_threshold = original_sensitivity_threshold;
+        }
+        return;
+      }
+      VibrationModeController *controller = getVibrationMode();
+
+      if (Config.autoEdgingDurationMinutes > 0 ) {   // Do the edging timer if not turned off
+        if ( millis() > (autoEdgingStartMillis + ( Config.autoEdgingDurationMinutes * 60 * 1000 )) && postOrgasmStartMillis == 0) {  // Detect if edging time has passed
+          Hardware::setEncoderColor(CRGB::Green);
+          if (Config.sensitivity_threshold != 6000) {
+            original_sensitivity_threshold = Config.sensitivity_threshold; // Make backup to bring back value after Post orgasm Torture
+            arousal = 0;                         //make sure arousal is lower then threshold bofore starting to detect an orgasm
+          }
+          Config.sensitivity_threshold = 6000; // make sure orgasm is now possible
```

Comment:
Changing the value like this might lead to the original getting lost, eg. some other config change saving the struct, and device resetting before this logic restores the value.

If your goal is instead to simply allow orgasm, you can achieve this by switching the device into Manual mode. This is done on the RunGraph page, I think you can call RunGraph.setMode(), check out how the Websocket server handles this. That way, you can guarantee that orgasm will happen because it stops tracking.

Additionally, if torture is the goal, you can add a new mode to the RunGraph page (this naming convention sucks, I know. Think of "Pages" like "apps" where "RunGraph" is the main edging "App" versus other "apps" you can run.) Anyway, you can add a new mode, instead of Automatic or Manual, call it Forced Orgasm, which when enabled will prevent the user from taking manual control of the device?